tests/compile/pxd_mangling_names.srctree

```diff
@@ -0,0 +1,43 @@
+# mode: compile
+# ticket: 2940
+
+PYTHON setup.py build_ext --inplace
+#PYTHON -c "import a; a.test()"
+
+######## setup.py ########
+
+from Cython.Build import cythonize
+from Cython.Distutils.extension import Extension
+from distutils.core import setup
+
+setup(
+    ext_modules=cythonize([Extension("a", ["a.py", "b.c"])]),
+)
+
+######## a.pxd ########
+
+cdef public int foo()
+
+cdef extern from "b.h":
+    cpdef int bar()
+
+######## a.py ########
+
+def foo():
+    return 42
+
+def test():
+    assert bar() == 42
+
+######## b.h ########
+
+#pragma once
+
+int bar();
```

Comment:
```suggestion
#ifndef A_H
#define A_H

int bar();

#endif
```

I'd rather keep this compatible as possible if there isn't a good reason to do otherwise